sdk/bootstrap/main.go

```diff
@@ -80,29 +85,40 @@ func parseArgs() args {
 	flag.BoolVar(&args.installCerts, "install-certs", true,
 		"Whether to install certs from .ssl to the JRE.")
 
-	flag.BoolVar(&args.getTaskIp, "get-task-ip", false, "Print task IP")
+	flag.BoolVar(&args.getTaskIP, "get-task-ip", false, "Print task IP")
 
 	flag.Parse()
 
 	// Note: Parse this argument AFTER flag.Parse(), in case user is just running '--help'
-	if args.resolveEnabled && rawHosts == defaultHostString && !args.getTaskIp {
+	if args.resolveEnabled && rawHosts == defaultHostString && !args.getTaskIP {
 		// Note: only build the default resolve value (requiring envvars) *after* we know
 		// the user didn't provide hosts of their own.
-		taskName, taskNameOk := os.LookupEnv("TASK_NAME")
-		frameworkHost, frameworkHostOk := os.LookupEnv("FRAMEWORK_HOST")
-		if !taskNameOk || !frameworkHostOk {
+		taskHost, err := parseTaskHost()
+		if err != nil {
 			printEnv()
 			log.Fatalf("Missing required envvar(s) to build default -resolve-hosts value. " +
 				"Either specify -resolve-hosts or provide these envvars: TASK_NAME, FRAMEWORK_HOST.")
+
 		}
-		args.resolveHosts = []string{fmt.Sprintf("%s.%s", taskName, frameworkHost)}
+
+		args.resolveHosts = []string{taskHost}
 	} else {
 		args.resolveHosts = splitAndClean(rawHosts, ",")
 	}
 
 	return args
 }
 
+func parseTaskHost() (string, error) {
+	taskName, taskNameOk := os.LookupEnv("TASK_NAME")
+	frameworkHost, frameworkHostOk := os.LookupEnv("FRAMEWORK_HOST")
+	if !taskNameOk || !frameworkHostOk {
+		return "", fmt.Errorf("Cannot determine TASK_NAME or FRAMEWORK_HOST")
```

Comment:
Could have this return the message currently at L99/L219, and then have L99/L219 log whatever's in `err`

Reply:
The message is slightly different:

@99: "set resolve hosts"

@219: "Time ran out waiting to resolve X to Y"